### PR TITLE
Add TVL metrics

### PR DESCRIPTION
**Pull request by michalsidzej**
Resolves L2B-580

### CODE REVIEW

File: packages/common/src/tools/queue/TaskQueue.ts [hunk 1]
```diff
@@ -153,3 +160,15 @@ export class TaskQueue<T> {
     }
   }
 }
+
+function wrapAndMeasure<T extends unknown[], U>(
```
Comment (krzkaczor) [inline packages/common/src/tools/queue/TaskQueue.ts:164]: i wonder can this be shared across this and BaseRepository's wrap

File: packages/backend/src/Metrics.ts [hunk 2]
```diff
@@ -58,6 +60,13 @@ export class Metrics {
         'uncertaintyBuffer',
       ],
     })
+
+    this.tvlHistogram = this.createHistogram({
+      name: 'tvl_sync_histogram',
```
Comment (krzkaczor) [inline packages/backend/src/Metrics.ts:65]: I think this deserves a better description. 
`tvl_sync_histogram` -> `tvl_sync_duration_histogram`
`Histogram showing tvl core sync speed` -> `Histogram showing TVL sync duration`

File: packages/backend/src/core/reports/ReportUpdater.ts [hunk 3]
```diff
@@ -26,12 +27,19 @@ export class ReportUpdater {
...
     private readonly logger: Logger,
+    metrics: Metrics,
   ) {
     this.logger = this.logger.for(this)
     this.configHash = getReportConfigHash(projects)
     this.taskQueue = new TaskQueue(
       (timestamp) => this.update(timestamp),
       this.logger.for('taskQueue'),
+      {
+        metrics: {
```
Comment (krzkaczor) [inline packages/backend/src/core/reports/ReportUpdater.ts:38]: Maybe we need something like `this.metrics.for(this)`.

File: packages/common/src/tools/queue/TaskQueue.ts (hunk 1 above)
Comment (michalsidzej) [reply packages/common/src/tools/queue/TaskQueue.ts:164]: That was probably my goal looking at the generic types here. Will do

**Review by krzkaczor — APPROVED** [review]
Accepted but this should be refactored in the future.

**Pull request merged**